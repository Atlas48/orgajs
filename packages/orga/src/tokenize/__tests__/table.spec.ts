import { tokenize } from '../index'

describe('tokenize table', () => {

  it('knows table hr', () => {
    expect(tokenize('|----+---+----|').all()).toMatchSnapshot()
    expect(tokenize('|--=-+---+----|').all()).toMatchSnapshot()
    expect(tokenize('  |----+---+----|').all()).toMatchSnapshot()
    expect(tokenize('|----+---+----').all()).toMatchSnapshot()
    expect(tokenize('|---').all()).toMatchSnapshot()
    expect(tokenize('|-').all()).toMatchSnapshot()
  })

  it('knows these are not table separators', () => {
    expect(tokenize('----+---+----|').all()).toMatchSnapshot()
  })

  it('knows table rows', () => {
    expect(tokenize('| batman | superman | wonder woman |').all()).toMatchSnapshot()
    expect(tokenize('| hello | world | y\'all |').all()).toMatchSnapshot()
    expect(tokenize('   | hello | world | y\'all |').all()).toMatchSnapshot()
    expect(tokenize('|    hello |  world   |y\'all |').all()).toMatchSnapshot()
    // with empty cell
    expect(tokenize('||  world   | |').all()).toMatchSnapshot()
  })

  it('knows these are not table rows', () => {
    expect(tokenize(' hello | world | y\'all |').all()).toMatchSnapshot()
    expect(tokenize('|+').all()).toMatchSnapshot()
  })
})
