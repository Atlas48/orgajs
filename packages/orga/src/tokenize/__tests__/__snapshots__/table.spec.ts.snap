// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize table knows table hr 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table hr 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table hr 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 17,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table hr 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table hr 5`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table hr 6`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.hr",
  },
]
`;

exports[`tokenize table knows table rows 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 9,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " batman ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 9,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 10,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " superman ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 21,
        "line": 0,
      },
      "start": Object {
        "column": 20,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 35,
        "line": 0,
      },
      "start": Object {
        "column": 21,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " wonder woman ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 36,
        "line": 0,
      },
      "start": Object {
        "column": 35,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
]
`;

exports[`tokenize table knows table rows 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 8,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " hello ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 9,
        "line": 0,
      },
      "start": Object {
        "column": 8,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 16,
        "line": 0,
      },
      "start": Object {
        "column": 9,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " world ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 17,
        "line": 0,
      },
      "start": Object {
        "column": 16,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 24,
        "line": 0,
      },
      "start": Object {
        "column": 17,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " y'all ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 25,
        "line": 0,
      },
      "start": Object {
        "column": 24,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
]
`;

exports[`tokenize table knows table rows 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 3,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 4,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " hello ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 19,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " world ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 19,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 27,
        "line": 0,
      },
      "start": Object {
        "column": 20,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": " y'all ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 28,
        "line": 0,
      },
      "start": Object {
        "column": 27,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
]
`;

exports[`tokenize table knows table rows 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "    hello ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "  world   ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 23,
        "line": 0,
      },
      "start": Object {
        "column": 22,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 29,
        "line": 0,
      },
      "start": Object {
        "column": 23,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "y'all ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 30,
        "line": 0,
      },
      "start": Object {
        "column": 29,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
]
`;

exports[`tokenize table knows table rows 5`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "  world   ",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 14,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
]
`;

exports[`tokenize table knows these are not table rows 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 24,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "hello | world | y'all |",
  },
]
`;

exports[`tokenize table knows these are not table rows 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "table.columnSeparator",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "+",
  },
]
`;

exports[`tokenize table knows these are not table separators 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "----+---+----|",
  },
]
`;
